website_event_require_login/controllers/main.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.http import request, route
+
+from odoo.addons.website_event.controllers.main import WebsiteEventController
+
+
+class RequireLoginToRegister(WebsiteEventController):
+    @route()
+    def registration_new(self, event, **post):
+        public_user = request.env.user == request.website.user_id
+        if public_user and event.website_require_login:
+            return request.env["ir.ui.view"].render_template(
```

Comment:
I think the UI and code could be simplified.

1. Replace the event registration button by a "Login to register" button, which leads directly to `/web/login?whatever`.
1. Once logged in, it returns to the event registration form, but the register button is available now.